src/components/hint/index.tsx

```diff
@@ -32,127 +34,169 @@ const TARGET_POSITIONS = {
+  left?: number;
+  right?: number;
+}
+
+interface HintPositionStyle extends Position {
+  alignItems?: string;
+}
+
+interface Paddings {
+  paddingLeft?: number;
+  paddingRight?: number;
+  paddingVertical?: number;
+  paddingHorizontal?: number;
+}
+
+export interface HintProps {
+  /**
+    * Control the visibility of the hint
+    */
+  visible?: boolean;
+  /**
+    * The hint background color
+    */
+  color?: string;
+  /**
+    * The hint message
+    */
+  message?: string | ReactElement;
+  /**
+    * The hint message custom style
+    */
+  messageStyle?: StyleProp<TextStyle>;
+  /**
+    * Icon to show next to the hint's message
+    */
+   icon?: ImageSourcePropType;
+  /**
+    * The icon's style
+    */
+   iconStyle?: StyleProp<ImageStyle>;
+  /**
+    * The hint's position
+    */
+   position?: HintPositions;
+  /**
+    * Provide custom target position instead of wrapping a child
+    */
+   targetFrame?: HintTargetFrame;
+  /**
+    * Show side tips instead of the middle tip
+    */
+   useSideTip?: boolean;
+  /**
+    * The hint's border radius
+    */
+   borderRadius?: number;
+  /**
+    * Hint margins from screen edges
+    */
+   edgeMargins?: number;
+  /**
+    * Hint offset from target
+    */
+   offset?: number;
+  /**
+    * Callback for the background press
+    */
+   onBackgroundPress?: (event: GestureResponderEvent) => void;
+  /**
+    * The hint container width
+    */
+   containerWidth?: number;
+  /**
+    * The hint's test identifier
+    */
+   testID?: string;
+  /**
+   * Additional styling
+   */
+  style?: StyleProp<ViewStyle>;
+}
+
+interface HintState {
+  targetLayout?: HintTargetFrame;
+  targetLayoutInWindow?: HintTargetFrame;
+}
+
+class Hint extends Component<HintProps, HintState> {
+  static displayName = 'Hint';
 
   static defaultProps = {
-    position: HINT_POSITIONS.BOTTOM
+    position: HintPositions.BOTTOM
   };
 
-  static positions = HINT_POSITIONS;
+  static positions = HintPositions;
+
+  private targetRef?: any;
+  private hintRef?: any;
 
   state = {
+    targetLayoutInWindow: undefined,
     targetLayout: this.props.targetFrame
   };
 
   visibleAnimated = new Animated.Value(Number(!!this.props.visible))
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps: HintProps) {
     if (prevProps.visible !== this.props.visible) {
       Animated.timing(this.visibleAnimated, {
         toValue: Number(!!this.props.visible),
-        duration: 170
+        duration: 170,
+        useNativeDriver: true
       }).start();
     }
   }
 
   focusAccessibilityOnHint = () => {
     const {message} = this.props;
-    const targetRefTag = findNodeHandle(this.targetRef);
-    const hintRefTag = findNodeHandle(this.hintRef);
-    if (targetRefTag && _.isString(message)) {
-      AccessibilityInfo.setAccessibilityFocus(targetRefTag);
-    } else if (hintRefTag) {
-      AccessibilityInfo.setAccessibilityFocus(hintRefTag);
+    if (this.targetRef && this.hintRef) {
```

Comment:
`targetRef` and `hintRef` are undefined at the start by definition, but you're not getting errors because the type is `any`, which is not great, I know. I found a potential solution for this problem though. Will push the changes soon.